src/runtime/rpc/rpc_base.cc

```diff
@@ -0,0 +1,166 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file rpc_base.cc
+ * \brief Base definitions for RPC.
+ */
+#include <iostream>
+#include <set>
+#include <chrono>
+#include <thread>
+#include <vector>
+#include <string>
+#include <random>
+
+#include "../../common/socket.h"
+#include "rpc_base.h"
+
+namespace tvm {
+namespace runtime {
+
+/*!
+ * \brief Send the data to remote.
+ * \param sock The socket.
+ * \param data The data to be sent.
+ */
+void SendData(common::TCPSocket sock, std::string data) {
+  int datalen = data.length();
+  CHECK_EQ(sock.SendAll(&datalen, sizeof(datalen)), sizeof(datalen));
+  CHECK_EQ(sock.SendAll(data.c_str(), datalen), datalen);
+}
+
+/*!
+ * \brief Receive the data to remote.
+ * \param sock The socket.
+ * \return The data received.
+ */
+std::string RecvData(common::TCPSocket sock) {
+  int datalen = 0;
+  CHECK_EQ(sock.RecvAll(&datalen, sizeof(datalen)), sizeof(datalen));
+  std::string data;
+  data.resize(datalen);
+  CHECK_EQ(sock.RecvAll(&data[0], datalen), datalen);
+  return data;
+}
+
+/*!
+* \brief Random Generate a random number between 0 and 1.
+* \return random float value.
+*/
+float Random() {
+  std::random_device rd;  // Will be used to obtain a seed for the random number engine
```

Comment:
A random number will only be useful if the device is persistent, instead of creating it every time.

Reply:
A correct way would be put the generator as member of the server class